### ISSUE TITLE
expose specific handshake parsers

### ISSUE THREAD

**Issue by GlenDC** [rusticata/tls-parser#66]
Hi. When trying to parse the client hello bytes as seen via Boring,
one gets the bytes specific to the client hello handshake message. At the moment one cannot parse this using `tls-parser` as the `parse_tls_handshake_msg_client_hello` parser function is not exposed.

Only option for now is to:

1. fork (which is what I did, but a bit of a shame)
2. prefix data with those few bytes hardcoded. This is easy enough, but not desired as you would need to allocate an entire vector just to hardcode these bytes... A lot of allocations for nothing

Would you be open to expose these individual parsers or what do you suggest?

**Comment by GlenDC** [comment]
FWIW I exposed the one fn I have to unblock me already in https://github.com/GlenDC/tls-parser/tree/feat/more-public-parsers, but probably that's not a true solution.

- might be that you want more functions to be public to also help others
- once these functions would be public might also make more sense to return the specific handshake data struct (e.g. client hello one) instead of still the all composing enum... As that anyway means I need to add a check for it, which feels silly as what else could it be given it succeeded and it called "parse client hello".

etc etc.. I am willing to drive this issue and fix, just need to have some guidance and know your perspective. If this is considered out of scope or not desirable for w/e reason, also fine. But please do let me know at your earliest convenience.

**Comment by chifflier** [comment]
Hi,
Sorry for the delay. If I understand, you would like to use the sub-parsers directly, like `parse_tls_handshake_msg_client_hello`.
Overall, I am in favor of this: the visibility change is not big, and it's a wanted feature for parser combinators to expose more parsers to be able to combine them.
I'll work on that for the next release.

The only thing I do not really like right now is the API, in particular the function names. That said, it is still a possibility to expose them now as experimental, and if this is required, transition to better naming.
